### PR TITLE
Convert more of tikv-server to generic kv engines

### PR DESCRIPTION
**Pull request by brson**
### What problem does this PR solve?

Issue Number: https://github.com/tikv/tikv/issues/10528 https://github.com/tikv/tikv/issues/6402

Problem Summary:

Continue making it possible to start tikv-server without rocks.

This PR is a partial rebase of https://github.com/tikv/tikv/pull/10529. That PR is bitrotted and needs to be reworked. This PR contains those commits that still work.

### What is changed and how it works?

Just refactoring to be generic over engines.

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/server/src/signal_handler.rs [hunk 1]
```diff
@@ -4,15 +4,14 @@ pub use self::imp::wait_for_signal;
...
-    use engine_traits::{Engines, MiscExt, RaftEngine};
+    use engine_traits::{Engines, KvEngine, MiscExt, RaftEngine};
     use libc::c_int;
     use nix::sys::signal::{SIGHUP, SIGINT, SIGTERM, SIGUSR1, SIGUSR2};
     use signal::trap::Trap;
     use tikv_util::metrics;
 
     #[allow(dead_code)]
-    pub fn wait_for_signal<ER: RaftEngine>(engines: Option<Engines<RocksEngine, ER>>) {
+    pub fn wait_for_signal<ER: RaftEngine>(engines: Option<Engines<impl KvEngine, ER>>) {
```
Comment (kennytm) [inline components/server/src/signal_handler.rs:14]: if the EK parameter is made `impl KvEngine` i think ER can be reduced to `impl RaftEngine` too? 🤔

File: components/server/src/signal_handler.rs [hunk 2]
```diff
@@ -37,8 +36,7 @@ mod imp {
 
 #[cfg(not(unix))]
 mod imp {
-    use engine_rocks::RocksEngine;
-    use engine_traits::Engines;
+    use engine_traits::{Engines, KvEngine, RaftEngine};
 
-    pub fn wait_for_signal(_: Option<Engines<RocksEngine, RocksEngine>>) {}
+    pub fn wait_for_signal<ER: RaftEngine>(_: Option<Engines<impl KvEngine, ER>>) {}
```
Comment (kennytm) [inline components/server/src/signal_handler.rs:41]: (ditto)

File: src/storage/config.rs [hunk 3]
```diff
@@ -105,20 +104,20 @@ impl Config {
...
-impl StorageConfigManger {
+impl<EK: KvEngine> StorageConfigManger<EK> {
     pub fn new(
-        kvdb: RocksEngine,
+        kvdb: EK,
         shared_block_cache: bool,
         ttl_checker_scheduler: Scheduler<TTLCheckerTask>,
         flow_controller: Arc<FlowController>,
-    ) -> StorageConfigManger {
+    ) -> StorageConfigManger<EK> {
```
Comment (kennytm) [inline src/storage/config.rs:120]:
```suggestion
    ) -> Self {
```

**Review by kennytm — COMMENTED** [review]
rest LGTM

**Review by kennytm — APPROVED** [review]

**Review by 5kbpers — APPROVED** [review]
LGTM

**Pull request merged**